script/utils/UpgradeTransparentProxy.s.sol

```diff
@@ -0,0 +1,123 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.19;
+
+import { JSONTimelockedOperations } from "../utils/JSONTimelockedOperations.sol";
+import { console2 } from "forge-std/console2.sol";
+import { TransparentUpgradeableProxy } from "@openzeppelin/contracts/proxy/transparent/TransparentUpgradeableProxy.sol";
+import { ProxyAdmin } from "@openzeppelin/contracts/proxy/transparent/ProxyAdmin.sol";
+import { TimelockController } from "@openzeppelin/contracts/governance/TimelockController.sol";
+import { EIP1967Helper } from "../utils/EIP1967Helper.sol";
+
+/// @notice Helper script that generates a json file with the timelocked operation to upgrade a TransparentUpgradeableProxy
+/// @dev Set in the constructor Modes.SCHEDULE to run _scheduleActions, Modes.EXECUTE to run _executeActions
+/// or Modes.CANCEL to run _cancelActions
+abstract contract UpgradeTransparentProxy is JSONTimelockedOperations {
+    // The address of the sender with proposer role
+    address public from;
+    // The address of the proxy to upgrade
+    address public proxyAddress;
+    // The address of the new implementation
+    address public newImplementationAddress;
+    // The proxy admin address that will be called
+    address public proxyAdminAddress;
+    // The operation salt for deterministic operation ID
+    bytes32 public salt;
+
+    constructor(address _proxyAddress, address _newImplementationAddress, string memory message, Modes _mode, address _timelock, address _from) JSONTimelockedOperations(
+        message,
+        _mode,
+        _timelock
+    ) {
+        from = _from;
+        proxyAddress = _proxyAddress;
+        newImplementationAddress = _newImplementationAddress;
+
+        // Get the proxy admin address from the proxy
+        proxyAdminAddress = EIP1967Helper.getAdmin(proxyAddress);
+        console2.log("ProxyAdmin address:", proxyAdminAddress);
+        
+        // Create a deterministic salt
+        salt = keccak256("salt");
```

Comment:
any chance we want to make this salt dynamically set?

Reply:
Salts are only required when we need to schedule the same action twice (same target, same params).
Parameter is not important here